bip-psbt.mediawiki

```diff
@@ -0,0 +1,245 @@
+<pre>
+  BIP: PSBT
+  Layer: Applications
+  Title: Partially Signed Bitcoin Transaction Format
+  Author: Andrew Chow <achow101@gmail.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-PSBT
+  Status: Draft
+  Type: Standard
+  Created: 2017-07-12
+  License: BSD-2-Clause
+</pre>
+
+==Introduction==
+
+===Abstract===
+
+This document proposes a binary transaction format which contains the information
+necessary for a signer to produce signatures for the transaction and holds the
+signatures for an input while the input does not have a complete set of signatures.
+
+===Copyright===
+
+This BIP is licensed under the 2-clause BSD license.
+
+===Motivation===
+
+Creating unsigned or partially signed transactions to be passed around to multiple
+signers is currently implementation dependent, making it hard for people who use
+different wallet software from being able to easily do so. One of the goals of this
+document is to create a standard format that can be used between clients to allow
+people to pass around the same transaction to sign and combine their signatures.
+
+Signing transactions also requires users to have access to the blockchain in order
+to find the UTXOs being spent so that the transaction can be signed. This transaction
+format will allow offline signers such as air-gapped wallets and hardware wallets
+to be able to sign transactions without access to the blockchain.
+
+==Specification==
+
+The transaction format uses a key-value map. Each key-value pair is as follows:
+
+{| class="wikitable" style="width: auto; text-align: center; font-size: smaller; table-layout: fixed;"
+!Name
+!Type
+!Description
+|-
+| Key Length
+| Compact Size Unsigned Integer
+| Specify how long the key is
+|-
+| Key
+| byte[]
+| The key itself
+|-
+| Value Length
+| Compact Size Unsigned Integer
+| Specify how long the value is
+|-
+| Value
+| byte[]
+| The Value itself
+|}
+
+The first byte of each key specifies the type of the key-value pair. The currently
+defined types are as follows:
+
+{| class="wikitable" style="width: auto; text-align: center; font-size: smaller;
```

Comment:
Perhaps split this table into "global" fields (transaction, redeemscript, witness script), and "per-input" fields (UTX, UTXO, partial sig). You can then later refer to them in the overall transaction table as just "array of global fields" and "array of array of local fields, with 0x00 separators" or so (rather than partially repeating the description for the global fields).